modules/javafx.swing/src/main/java/javafx/embed/swing/JFXPanel.java

```diff
@@ -1037,15 +1106,15 @@ public void setEmbeddedScene(EmbeddedSceneInterface embeddedScene) {
                 return;
             }
             if (pWidth > 0 && pHeight > 0) {
-                scenePeer.setSize(pWidth, pHeight);
+                hScenePeer.setSize(pWidth, pHeight);
             }
-            scenePeer.setPixelScaleFactors((float) scaleFactorX, (float) scaleFactorY);
+            hScenePeer.setPixelScaleFactors((float) scaleFactorX, (float) scaleFactorY);
 
             invokeOnClientEDT(() -> {
-                dnd = new SwingDnD(JFXPanel.this, scenePeer);
+                dnd = new SwingDnD(JFXPanel.this, hScenePeer);
                 dnd.addNotify();
-                if (scenePeer != null) {
-                    scenePeer.setDragStartListener(dnd.getDragStartListener());
+                if (hScenePeer != null) {
```

Comment:
this null check can be removed, since the code will bail out on L1106 if `hScenePeer == null`

Reply:
ok